acceptance/support/datatable-utils.ts

```diff
@@ -0,0 +1,49 @@
+import {ConstraintViolation} from "../../src/lsrs/web/model/common";
+import {DirectoryModel, FileModel} from "../../src/lsrs/web/model/files";
+import DataRegistry from "./data-registry";
+import {Attribute} from "./test-constants";
+
+/**
+ * Converts a row of file model data table to FileModel object.
+ *
+ * @param fileModel file model data table row
+ */
+export const convertFileModel = (fileModel: string[]) => {
```

Comment:
This is an absolutely overkill of an idea, but a fun one nonetheless. I see that most of these methods are basically doing the following:
1. Pair object keys to array items
2. Optionally map the array item
3. Create an object from these key-value pairs

Which also means that technically you could create a method that takes the string array coming from the cucumber test, an array of object keys, and an array of mapping methods, zip all three of them into an array of triplets, apply the optional mapper on all triplets to reduce them to a pair, then feed those pairs to `Object.fromEntries`.

So this method would end up being this:
```ts
return convert(
    fileModel,
    ['reference', 'path', 'acceptedAs', 'description', 'originalFilename'],
    [s => s.replace(...)]
) as FileModel
```

Reply:
Hot damn, this is TS-poweruser solution. Looks sick, I'll give it a try, thanks!